packages/typespec-ts/test/modularIntegration/generated/lro/modular/src/StandardClient.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ClientOptions } from "./common/interfaces.js";
+import {
+  User,
+  createOrReplace,
+  beginCreateOrReplaceAndWait,
+  beginCreateOrReplace,
+  CreateOrReplaceOptions,
+  createStandard,
+  StandardContext
+} from "./api/index.js";
+import { OperationState, SimplePollerLike } from "@azure/core-lro";
+
+export class StandardClient {
+  private _client: StandardContext;
+
+  /** Illustrates bodies templated with Azure Core with long-running operation */
+  constructor(options: ClientOptions = {}) {
+    this._client = createStandard(options);
+  }
+
+  createOrReplace(
+    role: string,
+    name: string,
+    options: CreateOrReplaceOptions = { requestOptions: {} }
+  ): Promise<User> {
+    return createOrReplace(this._client, role, name, options);
+  }
+
+  /**
+   * Below code are manually written now and will be generated later
+   */
+  beginCreateOrReplaceAndWait(
+    role: string,
+    name: string,
+    options: CreateOrReplaceOptions = { requestOptions: {} }
+  ): Promise<User> {
+    return beginCreateOrReplaceAndWait(this._client, role, name, options);
+  }
+
+  beginCreateOrReplace(
```

Comment:
did you consider generating an overload that can resume from a serialized poller state? We have an example for one in @azure/ai-language-text.

Reply:
I'll take a look